### PR TITLE
[Fixes #9795] Registered members with edit/manage permission can not …

### LINKED ISSUE #9795: Registered members with edit/manage permission can not edit Resource
@ElenaGallo commented on [Thu Jul 28 2022](https://github.com/GeoNode/geonode-mapstore-client/issues/1121)

Registered members with edit/manage permission can not edit/manage Resource

**How to Reproduce**
- See the steps in the video below:

https://user-images.githubusercontent.com/56537133/181476655-c4fa681c-cb84-45fb-aa4e-90b3a5060c08.mov

### PR DESCRIPTION
**Pull request by mattiagiupponi**
…edit Resource

<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [ ] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

**Comment by giohappy** [comment]
@mattiagiupponi a test for the API is failing

**Review by giohappy — CHANGES_REQUESTED** [review]
https://github.com/GeoNode/geonode/pull/9803#issuecomment-1203753095

File: geonode/base/api/tests.py [hunk 1]
```diff
@@ -641,7 +641,7 @@ def test_write_resources(self):
             self.assertEqual('Foo Abstract', response.data['resource']['abstract'], response.data['resource']['abstract'])
             self.assertEqual('Foo Attribution', response.data['resource']['attribution'], response.data['resource']['attribution'])
             self.assertEqual('321-12345-987654321', response.data['resource']['doi'], response.data['resource']['doi'])
-            self.assertEqual(False, response.data['resource']['is_published'], response.data['resource']['is_published'])
+            self.assertEqual(True, response.data['resource']['is_published'], response.data['resource']['is_published'])
```
Comment (giohappy) [inline geonode/base/api/tests.py:644]: @mattiagiupponi if bobby is the owner he should be able to publish the resource, unless the advanced workflow is activated. So this asswertion should match `False`, isn't it?

Comment (mattiagiupponi) [reply geonode/base/api/tests.py:644]: As decided in the call, we set the `is_published` and `is_approved` to read_only params. This means that they cannot be updated

**Review by giohappy — APPROVED** [review]

**Pull request merged**